src/EnergyPlus/DXCoils.cc

```diff
@@ -15158,40 +15157,91 @@ void CalcTwoSpeedDXCoilStandardRating(EnergyPlusData &state, int const DXCoilNum
         }
 
         TargetNetCapacity = NetCapacityFactorPLTestPoint(PartLoadTestPoint) * NetCoolingCapRated;
-        std::array<Real64, 12> Par; // Parameter array passed to solver
-
-        Par[0] = double(DXCoilNum);
-        Par[1] = TempDryBulb_Leaving_Apoint;
-        Par[2] = TargetNetCapacity;
-        Par[3] = OutdoorUnitInletAirDryBulbTempPLTestPoint(PartLoadTestPoint);
-        Par[4] = CoolingCoilInletAirWetBulbTempRated;
-        Par[5] = CoolingCoilInletAirDryBulbTempRated;
-        Par[6] = NetCoolingCapRated;
+
+        // set up parameters for the solver here
+        Real64 const par3 = OutdoorUnitInletAirDryBulbTempPLTestPoint(PartLoadTestPoint);
+        Real64 par7 = FanPowerPerEvapAirFlowRate;
+        int fanInNode = 0;
+        int fanOutNode = 0;
+        Real64 externalStatic = 0.0;
+        int fanIndex = 0;
         if (state.dataDXCoils->DXCoil(DXCoilNum).RateWithInternalStaticAndFanObject) {
-            Par[7] = 0.0;
-            Par[8] = double(FanInletNode);
-            Par[9] = double(FanOutletNode);
-            Par[10] = ExternalStatic;
-            Par[11] = double(state.dataDXCoils->DXCoil(DXCoilNum).SupplyFanIndex);
-        } else {
-            Par[7] = FanPowerPerEvapAirFlowRate;
-            Par[8] = 0.0;
-            Par[9] = 0.0;
-            Par[10] = 0.0;
-            Par[11] = 0.0;
+            par7 = 0.0;
+            fanInNode = FanInletNode;
+            fanOutNode = FanOutletNode;
+            externalStatic = ExternalStatic;
+            fanIndex = state.dataDXCoils->DXCoil(DXCoilNum).SupplyFanIndex;
         }
 
         LowerBoundMassFlowRate = 0.01 * state.dataDXCoils->DXCoil(DXCoilNum).RatedAirMassFlowRate(1);
 
+        auto f = [&state, &DXCoilNum, &TempDryBulb_Leaving_Apoint, &TargetNetCapacity, &par3, &par7, &fanInNode, &fanOutNode, &externalStatic](
```

Comment:
This lambda is pretty large.  I trimmed the original residual function down pretty far before converting it to this lambda, but it's still large.  I could put it the contents back as a member function and then just call it in this lambda, passing arguments by name instead of the old Par payload way.